### PR TITLE
✨[darwin] add track segment meta

### PR DESCRIPTION
**Pull request by CaiJingLong**
Function introduction: Submitted for the metadata of slow-motion videos on the Darwin platform.
This function only supports iOS/macOS.

1. Obtain AVAsset corresponding to PHAsset
2. Obtain track information of AVAsset
3. Slice according to tracks and export timeMapping of each slice.

---
CheckList
- [x] Changelog
- [x] Readme
- [x] Docs for class

### CODE REVIEW

**Comment by CaiJingLong** [comment]
- [ ] Changelog
- [ ] Readme

**Comment by CaiJingLong** [comment]
The information obtained from a video I tested is as follows:

<details><summary>Time mapping json</summary>
<p>

```json
{
  "mediaType": "vide",
  "timeRange": {
    "startTime": 0.0,
    "duration": 30.915
  },
  "nominalFrameRate": 44.667640686035156,
  "segments": [
    {
      "source": {
        "startTime": 0.0,
        "duration": 0.9633333333333334
      },
      "target": {
        "startTime": 0.0,
        "duration": 0.9633333333333334
      }
    },
    {
      "source": {
        "startTime": 0.9633333333333334,
        "duration": 0.013333333333333334
      },
      "target": {
        "startTime": 0.9633333333333334,
        "duration": 0.015
      }
    },
    {
      "source": {
        "startTime": 0.9766666666666667,
        "duration": 0.03166666666666667
      },
      "target": {
        "startTime": 0.9783333333333334,
        "duration": 0.045
      }
    },
    {
      "source": {
        "startTime": 1.0083333333333333,
        "duration": 0.04666666666666667
      },
      "target": {
        "startTime": 1.0233333333333334,
        "duration": 0.08333333333333333
      }
    },
    {
      "source": {
        "startTime": 1.055,
        "duration": 0.05333333333333334
      },
      "target": {
        "startTime": 1.1066666666666667,
        "duration": 0.13
      }
    },
    {
      "source": {
        "startTime": 1.1083333333333334,
        "duration": 0.04833333333333333
      },
      "target": {
        "startTime": 1.2366666666666666,
        "duration": 0.18166666666666667
      }
    },
    {
      "source": {
        "startTime": 1.1566666666666667,
        "duration": 3.535
      },
      "target": {
        "startTime": 1.4183333333333332,
        "duration": 28.281666666666666
      }
    },
    {
      "source": {
        "startTime": 4.691666666666666,
        "duration": 0.028333333333333332
      },
      "target": {
        "startTime": 29.7,
        "duration": 0.10666666666666667
      }
    },
    {
      "source": {
        "startTime": 4.72,
        "duration": 0.03166666666666667
      },
      "target": {
        "startTime": 29.80666666666667,
        "duration": 0.075
      }
    },
    {
      "source": {
        "startTime": 4.751666666666667,
        "duration": 0.02666666666666667
      },
      "target": {
        "startTime": 29.881666666666668,
        "duration": 0.04833333333333333
      }
    },
    {
      "source": {
        "startTime": 4.778333333333333,
        "duration": 0.018333333333333333
      },
      "target": {
        "startTime": 29.93,
        "duration": 0.02666666666666667
      }
    },
    {
      "source": {
        "startTime": 4.796666666666667,
        "duration": 0.008333333333333333
      },
      "target": {
        "startTime": 29.956666666666667,
        "duration": 0.008333333333333333
      }
    },
    {
      "source": {
        "startTime": 4.805,
        "duration": 0.08
      },
      "target": {
        "startTime": 29.965,
        "duration": 0.07833333333333334
      }
    },
    {
      "source": {
        "startTime": 4.885,
        "duration": 0.8716666666666667
      },
      "target": {
        "startTime": 30.043333333333333,
        "duration": 0.8716666666666667
      }
    }
  ]
}
``` 

</p>
</details> 

The test file: https://www.aliyundrive.com/s/LoGVEX3p8pA

<details><summary>ffprobe of the slow video</summary>
<p>

```sh
ffprobe version 5.0.1 Copyright (c) 2007-2022 the FFmpeg developers
  built with Apple clang version 12.0.0 (clang-1200.0.32.29)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/5.0.1 --enable-shared --enable-pthreads --enable-version3 --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libbluray --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-librav1e --enable-librist --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libsoxr --enable-libzmq --enable-libzimg --disable-libjack --disable-indev=jack --enable-videotoolbox
  libavutil      57. 17.100 / 57. 17.100
  libavcodec     59. 18.100 / 59. 18.100
  libavformat    59. 16.100 / 59. 16.100
  libavdevice    59.  4.100 / 59.  4.100
  libavfilter     8. 24.100 /  8. 24.100
  libswscale      6.  4.100 /  6.  4.100
  libswresample   4.  3.100 /  4.  3.100
  libpostproc    56.  3.100 / 56.  3.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/Users/caijinglong/Downloads/IMG_0023.MOV':
  Metadata:
    major_brand     : qt
    minor_version   : 0
    compatible_brands: qt
    creation_time   : 2023-03-25T01:22:42.000000Z
    com.apple.quicktime.location.accuracy.horizontal: 65.000000
    com.apple.quicktime.location.ISO6709: +39.8592+116.2596+054.317/
    com.apple.quicktime.make: Apple
    com.apple.quicktime.model: iPhone 7
    com.apple.quicktime.software: 14.2
    com.apple.quicktime.creationdate: 2023-03-25T09:18:55+0800
  Duration: 00:00:30.93, start: 0.000000, bitrate: 10259 kb/s
  Stream #0:0[0x1](und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, mono, fltp, 146 kb/s (default)
    Metadata:
      creation_time   : 2023-03-25T01:22:42.000000Z
      handler_name    : Core Media Audio
      vendor_id       : [0][0][0][0]
  Stream #0:1[0x2](und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, bt709, progressive), 1280x720, 10107 kb/s, 30 fps, 30 tbr, 600 tbn (default)
    Metadata:
      creation_time   : 2023-03-25T01:22:42.000000Z
      handler_name    : Core Media Video
      vendor_id       : [0][0][0][0]
      encoder         : H.264
    Side data:
      displaymatrix: rotation of -90.00 degrees
```

</p>
</details>

**Comment by CaiJingLong** [comment]
The recording time is 5 seconds, the last `source.start + source.duration` of the last segment.
And the video should be about 30 seconds.

File: example/lib/page/image_list_page.dart
Comment (AlexV525) [inline example/lib/page/image_list_page.dart:1]: Can we also add the meta from a real video to integrate with playback speed in order to see if we can handle them in handy?

**Pull request closed without merging**